### PR TITLE
Added disk-encryption-repart-dracut.service

### PR DESCRIPTION
**Pull request by schubi2**
before disk-encryption-dracut.service in order to repart the disk before encryption

### CODE REVIEW

File: disk-encryption-repart-dracut [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+{
+	echo "Error: $*" >&2
+	echo "Error: $*"
+	exit 1
+}
+
+####### main #######
+
+# silence systemd
+kill -SIGRTMIN+21 1
```
Comment (Vogtinator) [inline disk-encryption-repart-dracut:30]: If this is ment to write to stdout or such, using `StandardOutput=journal+console` should be enough

Comment (schubi2) [reply disk-encryption-repart-dracut]: Yes, that is meant. Great, that looks good !

**Comment by Vogtinator** [comment]
BTW, to work around the zypper issue that breaks the CI ATM you can apply this diff:

```diff
diff --git a/test/test.sh b/test/test.sh
index ad79585..d093f29 100755
--- a/test/test.sh
+++ b/test/test.sh
@@ -66,6 +66,7 @@ if ! [ -n "${reuseinitrd}" ] || ! [ -e "${tmpdir}/vmlinuz" ] || ! [ -e "${tmpdir
        rm -f "${tmpdir}/done"
        cat >create-initrd <<'EOF'
 #!/bin/bash
+# combustion: network
 set -euxo pipefail
 exec &>/dev/ttyS0
 trap '[ $? -eq 0 ] || poweroff -f' EXIT
diff --git a/test/testscript b/test/testscript
index 59b3034..bb23e2a 100644
--- a/test/testscript
+++ b/test/testscript
@@ -1,5 +1,5 @@
 #!/bin/bash
-# combustion: prepare
+# combustion: prepare network
 set -euxo pipefail
 exec &>/dev/ttyS0
 
```

**Comment by schubi2** [comment]
> Do you have more info what this PR does, like a usage example?

It calls systemd-repart before d-e-t because d-e-t enlarge the disk to the maximum. So, systemd-repart has no left spaces if it is called after d-e-t. 
The first try has been to include the systemd-repart call in the d-e-t service what Alberto has not really liked. :-)
Thorsten and Alberto has made the  suggestion to put it in an own service and call it just before d-e-t.

**Comment by Vogtinator** [comment]
> > Do you have more info what this PR does, like a usage example?
> 
> It calls systemd-repart before d-e-t because d-e-t enlarge the disk to the maximum. So, systemd-repart has no left spaces if it is called after d-e-t. The first try has been to include the systemd-repart call in the d-e-t service what Alberto has not really liked. :-) Thorsten and Alberto has made the suggestion to put it in an own service and call it just before d-e-t.

Not sure whether this is the right place to discuss the design, but what about the even simpler option of not having d-e-t enlarge at all and let systemd-repart do it normally? That way we can use the upstream integration unmodified, which avoids issues like it running twice on boot...

**Comment by aplanas** [comment]
> Not sure whether this is the right place to discuss the design, but what about the even simpler option of not having d-e-t enlarge at all and let systemd-repart do it normally? 

Re encryption is slow and the resize makes a lot of sense. The problem is that is not resizing back to the same original size. This has an easy solution and IMHO we should do that.

That way systemd-repart can work later after the switch root and ...

> That way we can use the upstream integration unmodified, which avoids issues like it running twice on boot...

**Comment by schubi2** [comment]
> BTW, to work around the zypper issue that breaks the CI ATM you can apply this diff:
> 
> ```diff
> diff --git a/test/test.sh b/test/test.sh
> index ad79585..d093f29 100755
> --- a/test/test.sh
> +++ b/test/test.sh
> @@ -66,6 +66,7 @@ if ! [ -n "${reuseinitrd}" ] || ! [ -e "${tmpdir}/vmlinuz" ] || ! [ -e "${tmpdir
>         rm -f "${tmpdir}/done"
>         cat >create-initrd <<'EOF'
>  #!/bin/bash
> +# combustion: network
>  set -euxo pipefail
>  exec &>/dev/ttyS0
>  trap '[ $? -eq 0 ] || poweroff -f' EXIT
> diff --git a/test/testscript b/test/testscript
> index 59b3034..bb23e2a 100644
> --- a/test/testscript
> +++ b/test/testscript
> @@ -1,5 +1,5 @@
>  #!/bin/bash
> -# combustion: prepare
> +# combustion: prepare network
>  set -euxo pipefail
>  exec &>/dev/ttyS0
>  
> ```

Hm, somehow it does not work.....

**Comment by Vogtinator** [comment]
> Hm, somehow it does not work.....

I think you didn't apply the diff correctly, it has to modify the embedded combustion script in test.sh. I suggest to copy-paste the diff into `git apply` (run that inside the repo root)

**Comment by schubi2** [comment]
> > Hm, somehow it does not work.....
> 
> I think you didn't apply the diff correctly, it has to modify the embedded combustion script in test.sh. I suggest to copy-paste the diff into `git apply` (run that inside the repo root)

Yes, my fault, sorry....

**Comment by aplanas** [comment]
Sorry for merging this: my bad. I think that I will revert it in #35 maybe keeping some fixes for the tests.

I think the plan should be something like:

1) d-e-t and systemd-repart are not related. There is no reason to add a service here for systemd-repart, and no reason for d-e-t to add systemd-repart in the initrd
2) To include systemd-repart in initrd we need to change the kiwi images, and add them in to the included dracut binaries
3) Also in the image we need to add an override in systemd-repart.service to add Before=disk-encryption-tool-dracut.service
4) The resize of d-e-t logic needs to be fixed, and revert to the original size

**Pull request merged**